grand/simulation/antenna/generic.py

```diff
@@ -139,22 +155,41 @@ def fftfreq(n, t):
         if dir_frame is not None:
             # Change the direction to the antenna frame
             if isinstance(direction, BaseRepresentation):
-                direction = dir_frame.realize_frame(direction)
-            direction = direction.transform_to(antenna_frame) # Now compute direction vector data in antenna frame
-            direction = direction.data
+                print("dir pre", direction, dir_frame, antenna_frame)
+                # LWP: this joins direction and dir_frame - not needed if we get rid of astropy
+                if grand_astropy:
+                    direction = dir_frame.realize_frame(direction)
+                else:
+                    E = field.E
+                
+                print("dir post", direction)
+                #exit()
+            
+            # LWP: This adds the difference of the bases (shower_frame-antenna_frame) to the direction, which can be done manually
+            if grand_astropy:
+                direction = direction.transform_to(antenna_frame).data
+                print("dire postpost", direction)
+            else:
+                # Bases are in meters
+                base_diff = numpy.array([dir_frame.location.x.value-antenna_frame.location.x.value, dir_frame.location.y.value-antenna_frame.location.y.value, dir_frame.location.z.value-antenna_frame.location.z.value])/1000
+                # LWP: The frames are NWU and ENU orientation, yet their values are as similar as they had the same orientation...
+                print(antenna_frame, dir_frame)
+                print(base_diff)
+                # LWP: ToDo: Need to check the orientations of frames and perhaps reposition the x,y,z
+                direction = numpy.array([direction.x.value,direction.y.value,direction.z.value])+base_diff
+                print(direction)
+                #exit()
 
         Leff:CartesianRepresentation
         Leff = self.model.effective_length(direction, f)
+
         if antenna_frame is not None:
             # Change the effective length to the E-field frame
             tmp = antenna_frame.realize_frame(Leff)
             tmp = tmp.transform_to(E_frame)
-            # Transorm from antenna_frame to E_frame is a translation.
-            # The Leff vector norm should therefeore not be modified, but it is because it is defined as a point in astropy coordinates
             Leff = tmp.cartesian
 
-        # Here we have to do an ugly patch for Leff values to be correct
```

Comment:
This was Olivier's hack in order to patch my bug. If astropy is removed this hack should not be needed anymore.